common-gitlab-cloudformation-utils/src/main/java/com/gitlab/aws/cfn/resources/shared/AbstractCombinedResourceHandler.java

```diff
@@ -221,9 +221,25 @@ public void update() throws Exception {
         public void delete() throws Exception {
             try {
                 Optional<ItemT> item = findExistingItemMatchingModel();
+                if(callbackContext != null) {
+                    // Delete has already been called and hopefully the project has now been deleted
+                    if (item.isPresent()) {
+                        // Deletion Failed or not yet done?
+                        if (callbackContext.getRetry() <= 0) {
+                            result = failure(HandlerErrorCode.NotStabilized, "Resource not deleted after 5 retries");
```

Comment:
Looks like the total retry time is going to be 10 seconds. Is that enough for stabilization?

Reply:
Stabalization seems to happen consistently in under 2 seconds.